### PR TITLE
MudCsvMapper #69

### PR DESCRIPTION
**Pull request by enkodellc**
#69 

This component is common in a lot of crm applications for importing files to existing tables. 

Could use some more UI work and looking forward to other suggestions.

### CODE REVIEW

**Comment by mckaragoz** [comment]
Thanks for the PR. I will review it as soon as possible.

**Comment by enkodellc** [comment]
No rush. I solved most of the issues with it and have it in production. It really could use an example with a table on it like you suggested. I could not find any other component set for blazor that has this component.

**Comment by mckaragoz** [comment]
Thats not an area i worked before, but i think i understand the logic.

There is a question: Why we build a component for this? What if we create only a class for mapping and just inject in a page? Then we can add this as utility service.

**Comment by enkodellc** [comment]
**Why we build a component for this?** Because people need them. Refer to the links below where other components like this were built in other languages.

**What if we create only a class for mapping and just inject in a page? Then we can add this as utility service.**  - A class for mapping might be fine but like the "SearchFunc" for the MudAutocomplete is done it could be supplied by the user and by default a simple string ignore case mapper is fine. I still think this should be a UI component utilizing both the drag and drop and file uploaded. 

Please review the following links for more context / references. I could not find a blazor one, that is why I think it should be done.

https://medium.com/better-through-code/building-a-csv-importer-part-1-8a016a470a23
https://github.com/jgile/vue-csv-import
https://github.com/beamworks/react-csv-importer

If you do not agree I will just host this component myself and cancel the PR. Other components as you see have 100+ stars so you can see others do use it.  This uses the MudBlazor existing components to create a new useful component which is what I see you already doing with MudExtensions.

**Comment by mckaragoz** [comment]
You are always welcome to create new components. As an extension package, its surely very good to have different utilities. I only ask for find the best way to build. As you see like MudWheel and MudSplitter we have different approaches to avoid js and make unique things.

So if its easier to with component, no problem for me. Then i ask you for some another things:

- We should make as simple as possible the code, name and API. `MudCsvMapper` sounds better i think. And in core library, Mud components has all Pascal case, for example its `MudFab` not `MudFAB` so `Csv` is better for Mud standard.

When you change the name and made your last controls, i can review all component and docs implementation.

**Comment by enkodellc** [comment]
@mckaragoz I refactored the name as requested. Please review.

File: CodeBeam.MudExtensions/Components/CsvMapper/MudCsvMapper.razor [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+@inherits MudComponentBase
+
+@using Microsoft.AspNetCore.Components
+@using System.ComponentModel.DataAnnotations
+@using Microsoft.AspNetCore.Components.Forms
+
+<div class="@Class" style="@Style">
+
+<style>
+	.mud-drop-item{
```
Comment (mckaragoz) [inline CodeBeam.MudExtensions/Components/CsvMapper/MudCsvMapper.razor]: Im not sure, but if a user uses another MudDropContainer in their page, doesn't this styles overrides the other container also? Maybe we can use `.mud-csv-mapper .mud-drop-item`

File: CodeBeam.MudExtensions/Components/CsvMapper/MudCsvMapper.razor [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+							<path d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM19 18H6c-2.21 0-4-1.79-4-4 0-2.05 1.53-3.76 3.56-3.97l1.07-.11.5-.95C8.08 7.14 9.94 6 12 6c2.62 0 4.88 1.86 5.39 4.43l.3 1.5 1.53.11c1.56.1 2.78 1.41 2.78 2.96 0 1.65-1.35 3-3 3zm-5.55-8h-2.9v3H8l4 4 4-4h-2.55z"></path>
+						</svg>
+					</MudPaper>
+					<MudPaper Elevation="0" class="d-flex justify-center align-content-center mud-transparent">
+						<h3>
+							<MudFab HtmlTag="Button"
+									Color="Color.Secondary"
+									Label="Choose a .csv file"
+									Icon="@Icons.Outlined.CloudDownload" Size="Size.Large"
+									for="@context" /> or Drag and Drop a CSV file here.
```
Comment (mckaragoz) [inline CodeBeam.MudExtensions/Components/CsvMapper/MudCsvMapper.razor]: The texts are hard coded. Does a user need localization? If true we may prepare a localized text parameter like we did in MudStepper.

**Comment by mckaragoz** [comment]
Thanks for the component, looking useful. Commented on some points, then we can merge.

File: CodeBeam.MudExtensions/Components/CsvMapper/MudCsvMapper.razor (hunk 1 above)
Comment (enkodellc) [reply CodeBeam.MudExtensions/Components/CsvMapper/MudCsvMapper.razor]: I agree I will fix.

**Comment by mckaragoz** [comment]
Thanks 🥇

**Pull request merged**